tg/update_handlers.py

```diff
@@ -27,7 +27,8 @@ def decorator(fun: UpdateHandler) -> UpdateHandler:
         @wraps(fun)
         def wrapper(controller: Controller, update: Dict[str, Any]) -> None:
             try:
-                fun(controller, update)
+                with controller.lock:
+                    fun(controller, update)
```

Comment:
Why do we need this lock? I think it is  redundant. All update are processed in sync, one event in the time, this code run in separate thread.

Reply:
btw, missing 2 change in one PR is not the best idea. lock vs idxs